pandas/util/_decorators.py

```diff
@@ -226,6 +226,105 @@ def wrapper(*args, **kwargs) -> Callable[..., Any]:
     return _deprecate_kwarg
 
 
+def _format_argument_list(allow_args: Union[List[str], int]):
+    """
+    Convert the allow_args argument (either string or integer) of
+    `deprecate_nonkeyword_arguments` function to a string describing
+    it to be inserted into warning message.
+
+    Parameters
+    ----------
+    allowed_args : list, tuple or int
+        The `allowed_args` argument for `deprecate_nonkeyword_arguments`,
+        but None value is not allowed.
+
+    Returns
+    -------
+    s : str
+        The substring describing the argument list in best way to be
+        inserted to the warning message.
+
+    Examples
+    --------
+    `format_argument_list(0)` -> ''
+    `format_argument_list(1)` -> 'except for the first argument'
+    `format_argument_list(2)` -> 'except for the first 2 arguments'
+    `format_argument_list([])` -> ''
+    `format_argument_list(['a'])` -> "except for the arguments 'a'"
+    `format_argument_list(['a', 'b'])` -> "except for the arguments 'a' and 'b'"
+    `format_argument_list(['a', 'b', 'c'])` ->
+        "except for the arguments 'a', 'b' and 'c'"
+    """
+    if not allow_args:
+        return ""
+    elif allow_args == 1:
+        return " except for the first argument"
+    elif isinstance(allow_args, int):
+        return " except for the first {num_args} arguments".format(num_args=allow_args)
+    elif len(allow_args) == 1:
+        return " except for the argument '{arg}'".format(arg=allow_args[0])
+    else:
+        last = allow_args[-1]
+        args = ", ".join(["'" + x + "'" for x in allow_args[:-1]])
+        return " except for the arguments {args} and '{last}'".format(
+            args=args, last=last
+        )
+
+
+def deprecate_nonkeyword_arguments(
+    version: str,
+    allowed_args: Optional[Union[List[str], int]] = None,
+    stacklevel: int = 2,
```

Comment:
Hmm why is stack level part of the signature here?

Reply:
I just done it like it was in the `deprecate_kwarg` decorator serving a close purpose.

Note that the stack level argument of `deprecate_kwarg` function currently seems to be never used, but if there is a reason to have it, it may be valueable here too.